### DISCUSSION TITLE
Add new category: Feature Flagging

### DISCUSSION THREAD

**Discussion by agardnerIT** [avelino/awesome-go#4883] [Q&A]
Hi, how would I add a new category for feature flagging? Specifically I want to submit the [OpenFeature](https://openfeature.dev) standard. Specifically for Go, I'd like to add a tutorial which uses [flagd](https://github.com/open-feature/flagd) (an open source OpenFeature compliant flag backend) + a snippet of Go code to show flag usage. Such as this:

## flags.json
```
{
    "flags": {
        "USE_LOG": {
            "state": "ENABLED",
            "variants": {
                "off": "false",
                "on": "true"
            },
            "defaultVariant": "on"
        }
    }
}
```

## Start flagd

```
docker run \
  --rm -it \
  --name flagd \
  -p 8013:8013 \
  -v $(pwd):/etc/flagd \
  ghcr.io/open-feature/flagd:latest start \
  --uri file:./flags.json

OR (run from the binary)
flagd start --port 8013 --uri file:flags.json
```

## main.go
`go run main.go`

```
package main

import (
	"context"
	"fmt"
	"time"

	flagd "github.com/open-feature/go-sdk-contrib/providers/flagd/pkg"
	"github.com/open-feature/go-sdk/pkg/openfeature"
)

func main() {

	flagd_host := "localhost"

	client := openfeature.NewClient("FeatureFlagSampleApp")

	openfeature.SetProvider(flagd.NewProvider(
		flagd.WithHost(flagd_host),
		flagd.WithPort(8013),
	))

	execution_counter := 1

	for {

		flag_value, _ := client.StringValue(
			context.Background(),
			"USE_LOG",
			"no",
			openfeature.EvaluationContext{},
		)

		fmt.Println("======================================")
		fmt.Printf("Execution Counter: %d \n", execution_counter)
		fmt.Printf("Feature flag value is: %s \n", flag_value)
		fmt.Println("To experiment: change the default_value in flags.json or")
		fmt.Println("Press Ctrl + C to exit")
		fmt.Println("======================================")

		execution_counter += 1
		time.Sleep(time.Second)
	}
}
```

The main purposes of this tutorial would be:

1) Introduce the concept of feature flagging
2) Explain that feature flags live outside of the code and thus can be updated, without deployment, in realtime.
3) Can lead to greater flexibility and business agility
4) Are thus (in some cases) better than using `.env` files and environment variables
5) Explain that a CNCF project called OpenFeature exists to standardise feature flagging between OS projects and any FF vendors

**Comment by avelino** [comment]
@agardnerIT have you read the contribution guide? there we talk about how to add a new category

> Categories must have at least 3 items.

https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#quality-standards
